map/framework.cpp

```diff
@@ -1739,6 +1723,38 @@ bool Framework::ShowMapForURL(string const & url)
   return false;
 }
 
+url_scheme::ParsingResult Framework::ParseApiURL(string const & url)
+{
+  using namespace url_scheme;
+  using namespace strings;
+
+  // clear every current API-mark.
+  {
+    UserMarkControllerGuard guard(m_bmManager, UserMarkType::API_MARK);
+    guard.m_controller.Clear();
+    guard.m_controller.SetIsVisible(true);
+    guard.m_controller.SetIsDrawable(true);
```

Comment:
Не понятно все равно.
Визибл растет из групп букмарок, которые можно по отдельности прятать. Больше отдельных групп ни в каких других типах юзермарков нет, просто показываем или прячем целиком.

> On 12.5.2016, at 08.52, Roman Kuznetsov <notifications@github.com> wrote:
> 
> In map/framework.cpp:
> 
> > @@ -1739,6 +1723,38 @@ bool Framework::ShowMapForURL(string const & url)
> >    return false;
> >  }
> >  
> > +url_scheme::ParsingResult Framework::ParseApiURL(string const & url)
> > +{
> > +  using namespace url_scheme;
> > +  using namespace strings;
> > +
> > +  // clear every current API-mark.
> > +  {
> > +    UserMarkControllerGuard guard(m_bmManager, UserMarkType::API_MARK);
> > +    guard.m_controller.Clear();
> > +    guard.m_controller.SetIsVisible(true);
> > +    guard.m_controller.SetIsDrawable(true);
> Насколько я помню IsDrawable означает, что группа юзер-марков (в данном случае API_MARK) должна отображаться в текущем состоянии программы. IsVisible говорит видима ли она конкретно сейчас. Разделение имеет практический смысл для букмарок, которые можно скрывать по группам.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>